src/Avalonia.Diagnostics/Diagnostics/Views/MainWindow.xaml.cs

```diff
@@ -18,6 +21,7 @@ namespace Avalonia.Diagnostics.Views
     internal class MainWindow : Window, IStyleHost
     {
         private readonly IDisposable _keySubscription;
+        private readonly Dictionary<Popup, IDisposable> _frozenPopupStates;
```

Comment:
I wonder if it can cause memory leaks. Should it be 
ConditionalWeakTable or do we clear this dictionary in time?

Reply:
It's probably better to dispose and clear all entries when DevTools close. Not sure what's exactly returned by `SetValue`, it probably has some sort of reference to the target object.